File: src/graphics/image_cropper/main_window.cpp
#include "main_window.h"
#include "ui_main_window.h"

#include <QFileDialog>
#include <QMessageBox>
#include <QPainter>
#include <QPainterPath>
#include <QWheelEvent>
#include <QtMath>
#include <QDebug>

MainWindow::MainWindow(QWidget *parent)
    : QMainWindow(parent), ui(new Ui::MainWindow) {
  ui->setupUi(this);
}

MainWindow::~MainWindow() { delete ui; }



void MainWindow::on_actionSave_triggered() {}

void MainWindow::on_actionOpen_triggered() {
  auto fileName =
      QFileDialog::getOpenFileName(this, tr("Open Image"), QDir::currentPath(),
                                   tr("Image Files (*.jpg *.JPG *.png *.PNG, *.bmp, *.BMP)"));

  ui->image_cropper->setPixmapFromFile(fileName);
}

File: src/graphics/python/image_cropper/main_window.py
from PyQt6 import QtWidgets, uic
from PyQt6.QtCore import QDir


class MainWindow(QtWidgets.QMainWindow):
    def __init__(self):
        super().__init__()
        uic.loadUi("main_window.ui", self)
        self.show()

    def on_actionSave_triggered(self):
        pass

    def on_actionOpen_triggered(self):
        file_name = QtWidgets.QFileDialog.getOpenFileName(
            self, "Open Image", QDir.currentPath(),
            "Image Files (*.jpg *.JPG *.png *.PNG, *.bmp, *.BMP)"
        )
        self.image_cropper.set_pixmap_from_file(file_name)
